src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -850,6 +851,18 @@ HTTPResponseWrapper run() {
           }
         }
       }
+    } else if (Set.of(HTTPMethod.DELETE, HTTPMethod.PATCH, HTTPMethod.PUT).contains(request.getMethod())) {
+      MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
```

Comment:
The reason this code adds this token on `POST is  to replicate what `prime-mvc` does in `AbstractControl.renderStart` which is used by the `form` control. 

See the comment on line `841`. 

So this code you are adding is now causing the simulator to do things that won't be done normally. So I would suggest you remove this code. 

Ideally the simulator would not do anything surprising, or different than the runtime in general. So in your case, you are adding a header during a `DELETE` request, but you are doing this manually. So as a result, if you want to test this behavior, you should also have to add the header in the test, and it should probably not occur in the simulator.